### PR TITLE
feat(examples): add tinygo postgres example

### PR DESCRIPTION
**Pull request by brooksmtownsend**
Signed-off-by: Brooks Townsend <brooksmtownsend@gmail.com>

## Feature or Problem
This PR adds a tinygo example for postgres, essentially mirroring the structure of the Rust example (just omitting the final query)

### CODE REVIEW

File: examples/golang/components/sqldb-postgres-query/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+With [wasmCloud][wasmcloud], you write only the important bits of your business logic, so the code for this component is short, with the important bits highlighted below:
+
+```go
+func (c Component) Call() string {
+    query := interfaces.WasmcloudPostgres0_1_0_draft_QueryQuery(CREATE_TABLE_QUERY, make([]PgValue, 0))
+    if query.IsErr() {
+        return fmt.Sprintf("ERROR: failed to create table: %v", query.UnwrapErr())
+    }
+    val := interfaces.WasmcloudPostgres0_1_0_draft_TypesPgValueText("inserted example go row!")
+    insertResult := interfaces.WasmcloudPostgres0_1_0_draft_QueryQuery(INSERT_QUERY, []PgValue{val})
```
Comment (vados-cosmonic) [inline examples/golang/components/sqldb-postgres-query/README.md:122]: Might be nice to have type aliases for some of the gnarly types here

File: examples/golang/components/sqldb-postgres-query/wadm.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  annotations:
+    version: v0.1.0
+    description: |
+      Demo WebAssembly component using the wasmCloud SQLDB Postgres provider via the wasmcloud:postgres WIT interface
+    wasmcloud.dev/authors: wasmCloud team
+    wasmcloud.dev/source-url: https://github.com/wasmCloud/wasmCloud/blob/main/examples/golang/components/sqldb-postgres-quer/wadm.yaml
+    wasmcloud.dev/readme-md-url: https://github.com/wasmCloud/wasmCloud/blob/main/examples/golang/components/sqldb-postgres-quer/README.md
+    wasmcloud.dev/homepage: https://github.com/wasmCloud/wasmCloud/tree/main/examples/golang/components/sqldb-postgres-quer
+    wasmcloud.dev/categories: |
+      database,sqldb,postgres,golang,example
```
Comment (vados-cosmonic) [inline examples/golang/components/sqldb-postgres-query/wadm.yaml:15]: this must be a bug elsewhere from copy paste but it `sqldb-postgres-query` is missing the `y`

File: examples/golang/components/sqldb-postgres-query/wadm.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+spec:
+  components:
+    - name: querier
+      type: component
+      properties:
+        image: file://./build/sqldb_postgres_query_s_s.wasm
```
Comment (vados-cosmonic) [inline examples/golang/components/sqldb-postgres-query/wadm.yaml:21]: Should this be double `_s_s`? Also, probably this should be `local.wadm.yaml`

We should also add this component to the example component checks -- the provider won't work but at least the component should come up

File: examples/golang/components/sqldb-postgres-query/wasmcloud.toml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name = "sqldb-postgres-query"
+version = "0.1.0"
+language = "tinygo"
+type = "component"
+
+[component]
+wit_world = "component"
```
Comment (vados-cosmonic) [inline examples/golang/components/sqldb-postgres-query/wasmcloud.toml:7]:
```suggestion
wit_world = "component"
```
I think you can exclude this -- though I'm open to it staying for clarity

File: examples/golang/components/sqldb-postgres-query/hello.go
Comment (vados-cosmonic) [inline examples/golang/components/sqldb-postgres-query/hello.go:1]: Do we want to call this `hello.go`?

**Review by vados-cosmonic — COMMENTED** [review]
Just a few comments -- mostly just want to get this under test like the other examples (to at least check component build & load)

**Review by vados-cosmonic — APPROVED** [review]

File: examples/golang/components/sqldb-postgres-query/README.md (hunk 1 above)
Comment (brooksmtownsend) [reply examples/golang/components/sqldb-postgres-query/README.md:122]: Yeah here the types are functions so it didn't work 😢 Unless there's a way to type alias the function?

File: examples/golang/components/sqldb-postgres-query/wadm.yaml (hunk 2 above)
Comment (brooksmtownsend) [reply examples/golang/components/sqldb-postgres-query/wadm.yaml:15]: whoops

File: examples/golang/components/sqldb-postgres-query/wadm.yaml (hunk 3 above)
Comment (brooksmtownsend) [reply examples/golang/components/sqldb-postgres-query/wadm.yaml:21]: Nope shouldn't be double `_s_s`

File: examples/golang/components/sqldb-postgres-query/wasmcloud.toml (hunk 4 above)
Comment (brooksmtownsend) [reply examples/golang/components/sqldb-postgres-query/wasmcloud.toml:7]: Apparently I do need it 😢 but perhaps we just need to adjust `wash build` for this?

```
missing `wit_world` in wasmcloud.toml ([component] section) for creating preview1 or preview2 components

```

**Comment by brooksmtownsend** [comment]
@vados-cosmonic lmk if this addressed your comments, good to merge once those are resolved!

File: examples/golang/components/sqldb-postgres-query/wadm.yaml (hunk 2 above)
Comment (vados-cosmonic) [reply examples/golang/components/sqldb-postgres-query/wadm.yaml]: yeah this is definitely my bug that carried over here, I think I went back and fixed it for the other one

File: examples/golang/components/sqldb-postgres-query/README.md (hunk 1 above)
Comment (vados-cosmonic) [reply examples/golang/components/sqldb-postgres-query/README.md:122]: Ahhhh Golang. Thanks for trying :(

**Comment by vados-cosmonic** [comment]
Sorry @brooksmtownsend a bit late here, but thanks for making those changes!

**Pull request merged**